### ISSUE TITLE
EC2 Credentials Provider not used on EC2 Mac instance

### ISSUE THREAD

**Issue by sebsto** [soto-project/soto-core#515]
**Describe the bug**
The default chain of credentials provider considers `.ec2` only when running on Linux.
Now that Amazon provides EC2 Mac instance, `.ec2` should be included in the default credential provider chain

**To Reproduce**
```
# create an AWSClient 
awsClient = AWSClient(
            credentialProvider: .default
            retryPolicy: .jitter(),
            httpClientProvider: .createNew)

# Compile and Run on EC2 Mac instance  (or compile on Mac laptop and run on EC2 Mac)
# with no `~/.aws` directory, just an IAM role attached to it

# It causes the following Error
SotoCore.CredentialProviderError
No credential provider found
```

**Expected behavior**
The AWSClient should pickup IAM meta data credentials by default

**Setup (please complete the following information):**
 - OS: 12.5.1
 - Version of Soto : latest
 - Authentication mechanism IAM Instance Profile on EC2 Mac

**Additional context**
The line that causes the problem is here https://github.com/soto-project/soto-core/blob/main/Sources/SotoCore/Credential/CredentialProvider.swift#L77

**Comment by adam-fowler** [comment]
I'm not sure this should be enabled by default.

You can already setup Soto to use an EC2 credential provider with the following
```swift
let client = AWSClient(credentialProvider: .ec2, ...)
```
If you want it to do the equivalent of what is the default for Linux you would provide this
```swift
let client = AWSClient(credentialProvider: .selector(.environment, .ecs, .ec2, .configfile()), ...)
```
This will go through the list of possible credential providers environment variables, ECS, EC2 and then the AWS config file until it finds one that works

**Comment by sebsto** [comment]
Agreed, this is what I do now, I moved away from the `.default` credentials provider. So this is not a blocker or a bug. I just found it surprising that the default behaviour changes depending on the OS.

There was a good reason for that when EC2 was Linux only (I am not even considering Windows here :-) But now that EC2 has macOS too, a library (or SDK) should not provide different behaviour for different OSes.

If you prefer to close this report and keep it like it is, I am good with that. Maybe just make it clear in the doc that macOS is not behaving the same as Linux.
